EIPS/eip-3416.md

```diff
@@ -0,0 +1,114 @@
+
+The transition to this gas price system will occur in two phases, in the first phase both legacy and EIP3416 transactions will be accepted by the protocol. Over the course of this first phase the amount of gas available for processing legacy transactions will decrease while the amount of gas available for processing EIP3416 transactions will increase, moving gas from the legacy pool into the EIP3416 pool until the legacy pool is depleted and the EIP3416 pool contains the entire gas maximum. After all of the gas has transitioned to the EIP3416 pool, the second, finalized, phase is entered and legacy transactions will no longer be accepted on the network.
+
+## Specification
+
+### Definitions
+
+This is a classic fork without a long migration time.
+
+* `FORK_BLOCK_NUMBER`: TBD.  Block number at or after which EIP-3416 transactions are valid.
+* `BLOCK_GAS_TARGET`: 10th item in the block header: The gas limit field, controlled by miner voting.  Previously referred to colloquially as `gas limit`, now referred to as `gas target`.  Controlled by miners in the same way as before where each miner can increase or decrease it by a very small amount relative to the parent block.
+* `CURRENT_BLOCK`: The current block that is being worked with (either being validated, or being produced).
+* `BASE_FEE`: State maintained by the miners. Represents the base minimum amount of nanoeth spent for every unit of gas a transaction uses. Not included as transaction data.
+* `BASE_FEE_MAX_CHANGE_DENOMINATOR`: `8`
+* `BASE_FEE_MAX_CHANGE`: `CURRENT_BLOCK.parent.BASE_FEE / BASE_FEE_MAX_CHANGE_DENOMINATOR`. The maximum amount the `BASE_FEE` can change between blocks (either up or down).
+* `LEGACY_GAS_PRICE`: Median gas price in `FORK_BLOCK_NUMBER - 1`.
+* `INITIAL_BASE_FEE` : `LEGACY_GAS_PRICE`.
+* `BASE_FEE_CHANGE`: `CURRENT_BLOCK.parent.BASE_FEE`. The amount that the BASE_FEE will increase in `CURRENT_BLOCK` relative to `CURRENT_BLOCK.parent`.
+
+### Process
+
+* At `block.number == INITIAL_FORK_BLOCK_NUMBER` we set `BASE_FEE = INITIAL_BASE_FEE`
+* `BASE_FEE` is set as follows
+  * Let `delta = block.gas_used - TARGET_GASUSED` (possibly negative).
+  * Set `BASE_FEE = PARENT_BASE_FEE + PARENT_BASE_FEE * delta // TARGET_GASUSED // BASE_FEE_MAX_CHANGE_DENOMINATOR`
+  * Clamp the resulting `BASE_FEE` inside of the allowable bounds if needed, where a valid `BASE_FEE` is one such that `abs(BASE_FEE - PARENT_BASE_FEE) <= max(1, PARENT_BASE_FEE // BASE_FEE_MAX_CHANGE_DENOMINATOR)`
+* EIP-3416 transactions are encoded the same as the current ones `rlp([nonce, gasPrice, gasLimit, to, value, data, v, r, s])` where `v,r,s` is a signature of `rlp([nonce, gasPrice, gasLimit, to, value, data])` and `gasPrice` is the `FEE_CAP` specified by the sender according to this proposal.
```

Comment:
Either there is no change in transactions at all, or there is a change in which case you'll need to specify a new transaction (e.g., an EIP-2718 transaction).  It sounds like you just want no new transaction type, in which case I recommend just leaving this sentence out and rewording some other stuff to make it clear that existing transactions are just being reinterpreted.

Reply:
I dont want any new Tx type as is not needed in this setting, and it also leave wallets without any need to upgrade. Will improve the redaction.